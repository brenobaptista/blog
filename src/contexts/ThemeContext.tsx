import { createContext } from 'react'
import { DefaultTheme } from 'styled-components'

import dracula from '@/styles/themes/dracula'

interface IThemeContext {
  theme: DefaultTheme
  toggleTheme: () => void
}

const ThemeContext = createContext<IThemeContext>({
  theme: dracula,
  toggleTheme: () => null
})

export default ThemeContext
