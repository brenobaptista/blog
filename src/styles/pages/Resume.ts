import styled, { createGlobalStyle } from 'styled-components'

export const PrintingRules = createGlobalStyle`
  @page {
    margin: 0;
    size: A4 portrait;
  }
`

export const Page = styled.div`
  color: #000000;
  background: #ffffff;
  width: 210mm;
  height: 297mm;
  padding: 25mm;

  @media screen {
    margin: 5mm auto;
  }
`

export const Content = styled.div`
  line-height: normal;
  font-size: 0.8em;
  text-align: justify;
`

export const Header = styled.header`
  display: flex;
  align-items: center;
  gap: 32px;
`

export const Author = styled.div`
  flex: 1;

  h1 {
    font-size: 1.3rem;
    font-weight: 700;
    margin: 0;
  }

  div {
    font-size: 1rem;
  }
`

export const Social = styled.div`
  a {
    display: flex;
    align-items: center;
    gap: 8px;
    margin: 4px 0;
    color: #000000;
    text-decoration: underline;
    cursor: pointer;
  }
`

export const Section = styled.div`
  h2 {
    font-size: 0.9rem;
    margin: 16px 0;
  }

  b,
  i {
    display: block;
  }

  div {
    margin-bottom: 16px;
  }
`
