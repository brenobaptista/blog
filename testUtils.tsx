import { render } from '@testing-library/react'
import { RenderResult } from '@testing-library/react/types'
import { ThemeProvider } from 'styled-components'

import light from './src/styles/themes/light'

const Providers = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
)

const customRender = (ui: React.ReactElement, options = {}): RenderResult =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
