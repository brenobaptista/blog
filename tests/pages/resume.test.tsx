import { render } from 'testUtils'
import Resume from '@/pages/resume'
import ThemeContext from '@/contexts/ThemeContext'

const renderWithContext = (ui, value) =>
  render(<ThemeContext.Provider value={value}>{ui}</ThemeContext.Provider>)

test('matches snapshot in dracula mode', () => {
  const theme = { mode: 'dracula' }
  const { asFragment } = renderWithContext(<Resume />, {
    theme
  })

  expect(asFragment()).toMatchSnapshot()
})
