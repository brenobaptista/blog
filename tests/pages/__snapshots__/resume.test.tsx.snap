// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot in dracula mode 1`] = `
<DocumentFragment>
  .c0 {
  color: #000000;
  background: #ffffff;
  width: 210mm;
  height: 297mm;
  padding: 10mm 15mm;
}

.c1 {
  line-height: normal;
  font-size: 0.7rem;
  text-align: justify;
}

.c1 a {
  color: #000000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 32px;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 h1 {
  font-size: 1.2rem;
  font-weight: 700;
  margin: 0;
}

.c3 div {
  font-size: 0.7rem;
}

.c4 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
  margin: 4px 0;
}

.c5 h2 {
  font-size: 0.8rem;
  margin: 24px 0 8px;
}

.c5 .mt-0 {
  margin-top: 0;
}

.c5 strong,
.c5 em {
  display: block;
}

.c5 ul,
.c5 ol {
  margin: 10px 0 20px;
  padding-left: 30px;
}

.c5 li {
  padding-bottom: 5px;
}

.c5 div {
  margin-bottom: 16px;
}

@media screen {
  .c0 {
    margin: 5mm auto;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        <div
          class="c3"
        >
          <h1>
            Breno Baptista
          </h1>
          <div>
            Software engineer with bachelorâ€™s degree in Computer Science and 4+ years experience at high-growth early-stage startups.
          </div>
        </div>
        <div
          class="c4"
        >
          <div>
            <a
              aria-label="Contact me by email"
              href="mailto:me@brenobaptista.com"
            >
              <svg
                aria-hidden="true"
                fill="none"
                height="16"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Contact me by email
                </title>
                <path
                  d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
                />
                <polyline
                  points="22,6 12,13 2,6"
                />
              </svg>
              <span>
                me@brenobaptista.com
              </span>
            </a>
          </div>
          <div>
            <a
              aria-label="My blog"
              href="https://www.brenobaptista.com/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                fill="none"
                height="16"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  RSS feed
                </title>
                <path
                  d="M4 4a16 16 0 0 1 16 16"
                />
                <path
                  d="M4 11a9 9 0 0 1 9 9"
                />
                <circle
                  cx="5"
                  cy="19"
                  r="1"
                />
              </svg>
              <span>
                brenobaptista.com
              </span>
            </a>
          </div>
          <div>
            <a
              href="https://github.com/brenobaptista/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                fill="none"
                height="16"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  GitHub profile
                </title>
                <path
                  d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                />
              </svg>
              <span>
                github.com/brenobaptista
              </span>
            </a>
          </div>
          <div>
            <a
              href="https://www.linkedin.com/in/breno-baptista/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                fill="none"
                height="16"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  LinkedIn profile
                </title>
                <path
                  d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
                />
                <rect
                  height="12"
                  width="4"
                  x="2"
                  y="9"
                />
                <circle
                  cx="4"
                  cy="4"
                  r="2"
                />
              </svg>
              <span>
                linkedin.com/in/breno-baptista
              </span>
            </a>
          </div>
        </div>
      </header>
      <div
        class="c5"
      >
        <h2
          class="mt-0"
        >
          EXPERIENCE
        </h2>
        <div>
          <strong>
            Software Engineer, Alternative Payments
          </strong>
          <em>
            February 2022 - Present (USA, remote)
          </em>
          <ul>
            <li>
              Provided an integrated B2B payments and checkout infrastructure for service-based businesses.
            </li>
            <li>
              Maintained highly-scalable and containerized microservices in Go. Developed the email service from scratch, ensuring reliability and seemless communication through RabbitMQ messages.
            </li>
            <li>
              Delighted clients by crafting web experiences. Translated design mockups into efficient UI components and responsive pages.
            </li>
          </ul>
        </div>
        <div>
          <strong>
            Full-Stack Developer, Massive
          </strong>
          <em>
            January 2020 - January 2022 (USA, remote)
          </em>
          <ul>
            <li>
              Fought intrusive advertising by offering monetization of apps without ads at the cost of unused CPU/GPU using cryptocurrencies, blockchain and web3 instead of personal attention and data.
            </li>
            <li>
              Led efforts to vastly reduce technical debt in web and desktop apps, launching a refactored partner dashboard with better UI/UX and performance.
            </li>
            <li>
              Helped businesses comply with data laws (GDPR/CCPA) with easy plugins by developing WordPress and Shopify plugins from scratch.
            </li>
          </ul>
        </div>
        <div>
          <strong>
            Product Owner, Acens
          </strong>
          <em>
            February 2019 - January 2020 (Brazil)
          </em>
          <ul>
            <li>
              Helped businesses build an online presence by managing a small team that developed high-impact projects to our users.
            </li>
            <li>
              Improved the organization productivity by enforcing agile across the company. Took part in critical product decisions during sprint meetings and calls with partners.
            </li>
          </ul>
        </div>
      </div>
      <div
        class="c5"
      >
        <h2>
          PROJECTS
        </h2>
        <ol>
          <li>
            <i>
              Programming Blog: 
            </i>
            <a
              href="https://github.com/brenobaptista/blog"
              rel="noreferrer noopener"
              target="_blank"
            >
              <span>
                github.com/brenobaptista/blog
              </span>
            </a>
          </li>
          <li>
            <i>
              Tailwind Dashboard Template: 
            </i>
            <a
              href="https://github.com/brenobaptista/plume-dashboard"
              rel="noreferrer noopener"
              target="_blank"
            >
              <span>
                github.com/brenobaptista/plume-dashboard
              </span>
            </a>
          </li>
          <li>
            <i>
              APR Service for Vehicle Loans: 
            </i>
            <a
              href="https://github.com/brenobaptista/apr-service"
              rel="noreferrer noopener"
              target="_blank"
            >
              <span>
                github.com/brenobaptista/apr-service
              </span>
            </a>
          </li>
          <li>
            <i>
              Deno URL Shortener Service: 
            </i>
            <a
              href="https://github.com/brenobaptista/deno-url-shortener"
              rel="noreferrer noopener"
              target="_blank"
            >
              <span>
                github.com/brenobaptista/deno-url-shortener
              </span>
            </a>
          </li>
        </ol>
      </div>
      <div
        class="c5"
      >
        <h2>
          EDUCATION
        </h2>
        <div>
          <strong>
            Bachelorâ€™s Degree in Computer Science
          </strong>
          <div>
            Unifacvest University Center (2019 - 2023)
          </div>
        </div>
        <div>
          <strong>
            Scrum Fundamentals Certified (SFCâ„¢)
          </strong>
          <div>
            SCRUMstudy - Accreditation Body for Scrum and Agile (2020)
          </div>
        </div>
      </div>
      <div
        class="c5"
      >
        <h2>
          SKILLS
        </h2>
        <div>
          <b>
            Languages: 
          </b>
          JavaScript, TypeScript, Go, Bash, Lua
        </div>
        <div>
          <b>
            Front-end: 
          </b>
          HTML, CSS, Sass, jQuery, Styled Components, Tailwind, React, React Native, Next.js, Storybook, Accessibility, Jest/RTL, Cypress
        </div>
        <div>
          <b>
            Back-end: 
          </b>
          Node, Express, Deno, Apollo, GraphQL, Postgres, MongoDB, Firebase
        </div>
        <div>
          <b>
            DevOps: 
          </b>
          Git, Docker, AWS, GNU/Linux, Yarn Workspaces, Turborepo, CI/CD
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
